### ISSUE TITLE
Redundant Transform Parameters

### ISSUE THREAD

**Issue by wmcginty** [BottleRocketStudios/iOS-Hyperspace#69]
After updating to Hyperspace 2.1, it feels like there is a lot of redundancy in the transform parameters.

```swift
    /// Attempts to parse the provided Data into the associated response model type for this request.
    ///
    /// - Parameter data: The raw Data retrieved from the network.
    /// - Parameter serviceSuccess: The successful result of executing a Request using a NetworkService.
    /// - Returns: A result indicating the successful or failed transformation of the data into the associated response type.
    func transformData(_ data: Data, serviceSuccess: NetworkServiceSuccess) -> Result<ResponseType, ErrorType>
```

The data is present as `Data`, `serviceSuccess.data`, and sometimes `serviceSuccess.response.data`. We should look to change the signature of the transform to:

```swift
func transformSuccess(_ serviceSuccess: NetworkServiceSuccess) -> Result<ResponseType, ErrorType>
```

This way we can unify the method name in the protocol with that of the default implementations. This also makes it much clearer when testing as you don't have to provide 'data' multiple times (where its possible each separate variable could be used independently depending on the implementation of `transform`).

Thoughts @tylermilner @earlgaspard  ?

**Comment by wmcginty** [comment]
It's non-optional in `NetworkServiceSuccess`. So as long as we don't rely on `HTTP.Response` data we should be able to rely on it's presence, no?

**Comment by earlgaspard** [comment]
Correct. I was thinking of HTTP.Response for some reason with it's optional data

**Comment by earlgaspard** [comment]
I was attempting to change things up in NetworkServiceHelper's networkServiceResult for the case .success:. There are times you could get back no data like a DELETE call and it would be fine to do so. I don't agree that it should result in returning a NetworkServiceFailure for that case. This is where issues happened for me when changed this functionality

**Comment by tylermilner** [comment]
I noticed this apparent redundancy too when I was reviewing some of the changes that went into `2.1.0`. I came to the same conclusion as @earlgaspard where there _seemed_ to be a valid reason for the way it's done now, or maybe we just weren't trying to change things up all over the place. I'll take another look.

**Comment by tylermilner** [comment]
I took a stab at how this would look in #70.

**Comment by wmcginty** [comment]
Damn Tyler, when I brought it up I didn't think you'd have it done a few hours later.

**Comment by tylermilner** [comment]
Don't worry, @wmcginty. I left you plenty of fun unit test and documentation updates for you! 😉

**Comment by wmcginty** [comment]
This can be closed as #70 fixes it.
